fix: flatten root filter group

almeidabbm opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

Closes: [#9537](https://github.com/lightdash/lightdash/issues/9537)

### Description:

- Flattens root filter group

**Before**

https://github.com/lightdash/lightdash/assets/22939015/c81ab972-068b-4fe3-821b-1fb74efff690

**After**

https://github.com/lightdash/lightdash/assets/22939015/79443612-e7f9-4cba-94fc-0504fa51c6cc

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

IrakliJani reviewed (commented):
@almeidabbm 
I think it would be best if we fixed this at the point where we build the underlying metric query, rather than after we've created the metric query. It seems like the problem is originating from there.

almeidabbm commented:
> @almeidabbm I think it would be best if we fixed this at the point where we build the underlying metric query, rather than after we've created the metric query. It seems like the problem is originating from there.

I changed the logic to where we determine the filters of the underlying metric query. Basically instead of flattening at the component level, we're now creating the filters object from the group instead of forcing an `AND` group in the dimensions filters by calling `getFiltersFromGroup` which also flattens the groups and removes any unnecessary nesting

IrakliJani reviewed (approved):
approving - assuming tests pass! well done 🚀
